src/viam/sdk/tests/CMakeLists.txt

```diff
@@ -13,14 +13,34 @@
 # limitations under the License.
 
 enable_testing()
+
 include(viamcppsdk_utils)
+add_library(viamsdk_test)
+
+target_sources(viamsdk_test
```

Comment:
This is the new `viamsdk_test` library that contains mocks and test utilities so they don't end up in the SDK library.